docs/develop/dotnet/message-passing.mdx

```diff
@@ -1,261 +1,791 @@
+
+### Set a dynamic Workflow {#set-a-dynamic-workflow}
+
+A dynamic Workflow refers to a special stand-in Workflow Definition.
+It's used when an unknown Workflow Execution request arrives.
+
+Consider the "MyUnknownWorkflow" example described earlier.
+The Worker may find there's no registered Workflow Definitions of that name or type.
+After failing to find a Workflow Definition with a matching type, the Worker looks for a dynamic stand-in.
+If found, it invokes that instead.
+
+To participate, your Workflow must opt into dynamic access.
+Adding `implements DynamicWorkflow` to the Workflow class declaration makes the Workflow Definition eligible to participate in dynamic invocation.
+You must register the Workflow with the [Worker](https://python.temporal.io/temporalio.worker.html) before it can be invoked.
+
+```csharp
+NEEDS EXAMPLE
+```
+
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**Needs .NET.**</div>
+The Workflow Definition's primary Workflow method must accept a single argument of type `EncodedValues`.
+Use a  [payload_converter](https://python.temporal.io/temporalio.workflow.html#payload_converter) function to convert `EncodedValues` objects to your required type.
+For example:
 
-**How to send an Update from a Temporal Client using the Temporal .NET SDK**
+<div style={{backgroundColor: '#e0e0e0',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}><a href="https://github.com/temporalio/documentation/blob/main/sample-apps/python/dynamic_handlers/your_dynamic_workflow_dacx.py"><tt>See full sample</tt></a></div>
 
-To send a Workflow Update from a Temporal Client, call the `ExecuteUpdateAsync` method on the `WorkflowHandle` class.
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**Needs .NET and an appropriate example with a payload converter.**</div>
 
 ```csharp
-var result = await handle.ExecuteUpdateAsync(wf => wf.UpdateStatusAsync(newStatus));
+ADD SAMPLE HERE LIKE PYTHON
 ```
+
+This Workflow converts the first element to a string, and uses that to execute an Activity.
+
+<div style={{backgroundColor: '#ffff00',padding: '0px 15px',borderRadius: '5px',border: '1px solid #cccccc',display: 'inline-block'}}>**.NET source did not include dynamic Activity coverage.**</div>
+
+### Set a dynamic Activity {#set-a-dynamic-activity}
+
+A dynamic Activity is a stand-in implementation.
+It's used when an Activity Task with an unknown Activity type is received by the Worker.
+
+To participate, your Activity must opt into dynamic access.
+Adding `implements DynamicActivity` to the type declaration makes the Activity Definition eligible to participate in dynamic invocation.
+You must register the Activity with the Worker before it can be invoked.
+
+```csharp
+NEEDS EXAMPLE
```

Comment:
Dynamic workflows and activities are with workflow and activity docs instead of with query, signal, update docs